beater/server.go

```diff
@@ -135,34 +128,23 @@ func processRequest(
 	buf, err := ioutil.ReadAll(limitedReader)
 	if err != nil {
 		// If we run out of memory, for example
-		return reportError(500, fmt.Errorf("Data read error: %s", err))
+		return 500, err
```

Comment:
You remove the additional error information here. In case this info is not already reflected in the original error (I haven't checked that) you could do something like 
`return 500, errors.New(fmt.Sprintf("xyz %s", err.Error()))`

Reply:
You mean the "Data read error" part? that was intended, for what i could see, errors are more or less self-descriptive (eg "unexpected EOF").
Do you think is needed to wrap it?

Reply:
no as I said, I did not look into the original error - if you checked, it is fine with me.

Reply:
The more descriptive errors were added on the one hand for the users but also to quickly nail down where the error exactly happened. `unexpected EOF` could happen in a few places.

Reply:
done 499ba3e31c72a174cfa12e4f69b2177e0a8e3668